commit c15e197ca00b2caad1abeac030d2e387b5fd32b2
Author: Luo Runbing <luo.runbing@zte.com.cn>
Date:   Mon Mar 1 09:00:13 2021 +0800

    mgr: create client messenger using ms_public_type
    
    and fallback to ms_type if ms_public_type is not configured
    
    otherwise rdma support breaks with ms_public_type set to async+rdma explicitly
    and left ms_type & ms_cluster_type to their defaults.
    
    Signed-off-by: luo.runbing <luo.runbing@zte.com.cn>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index 585837802f5..37a2219bccb 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -41,7 +41,8 @@ MgrStandby::MgrStandby(int argc, const char **argv) :
   monc{g_ceph_context, poolctx},
   client_messenger(Messenger::create(
 		     g_ceph_context,
-		     cct->_conf.get_val<std::string>("ms_type"),
+		     cct->_conf.get_val<std::string>("ms_public_type").empty() ?
+			cct->_conf.get_val<std::string>("ms_type") : cct->_conf.get_val<std::string>("ms_public_type"),
 		     entity_name_t::MGR(),
 		     "mgr",
 		     Messenger::get_pid_nonce())),
